### DISCUSSION TITLE
how to learn Node js

### DISCUSSION THREAD

**Discussion by Jamshdbek** [community/community#49902] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

How do you think how learn node js : tutorials , docs , courses     
Are there  effective routedmaps?

**Comment by maahaMaya** [comment]
Node is extension of JavaScript:
So first you learn [JavaScript](https://youtube.com/playlist?list=PLlasXeu85E9cQ32gLCvAvr9vNaUccPVNP)
https://www.w3schools.com/js/DEFAULT.asp

Then 
Follow
1. [https://www.w3schools.com/nodejs/nodejs_intro.asp](https://www.w3schools.com/nodejs/nodejs_intro.asp)
2. [https://www.javatpoint.com/nodejs-tutorial](https://www.javatpoint.com/nodejs-tutorial)
3. [https://youtu.be/ChVE-JbtYbM](https://youtu.be/ChVE-JbtYbM)

**Comment by Benyaak** [comment]
__Hello__,

If by __routedmaps__ you refer to __a learning roadmap__ that is recommended, there are a few resources that you can start with. I am going to list a few of those right below:

a. [Roadmap.sh](https://roadmap.sh/nodejs)
b. [Roadmap Helper](https://github.com/aliyr/Nodejs-Developer-Roadmap)
c. [NodeJs Medium Roadmap for 2023](https://javascript.plainenglish.io/nodejs-developer-roadmap-2023-4e2fccc0ff54)

But first things first, hopefully, you had the chance of learning javascript beforehand so that the courses presented in the roadmap make sense to you.

And about the efficient way of learning node, that is something you have to find out what best suits you. I personally learn the most efficiently with docs & hands-on learning courses.

**Reply by Jamshdbek** [reply]
thenks! for recommend and resources 👍

**Comment by jameshasung** [comment]
There are several resources available for learning Node.js, and the effectiveness of each may vary depending on your learning style and preferences. Here are a few options:

1. Tutorials: Node.js has a wealth of tutorials available online, including articles, blog posts, and video tutorials. These can be a great way to get started with Node.js and learn the basics. However, it's important to be aware that tutorials may not always be up-to-date or cover all of the topics you need to know.

2. Documentation: Node.js has comprehensive documentation available on its website, including guides, API reference, and more. The documentation is well-organized and regularly updated, making it a great resource for both beginners and more advanced users.

3. Courses: There are many online courses available for learning Node.js, ranging from free tutorials on platforms like YouTube and Udemy to more in-depth courses on platforms like Pluralsight and LinkedIn Learning. Courses can provide a structured learning experience and may be a good option if you prefer a more guided approach to learning.

As for "effective roadmap", there are several resources available online that provide a structured learning path for Node.js. For example, the Node.js website offers a learning path for beginners, which includes several tutorials and guides covering the basics of Node.js. Additionally, platforms like Pluralsight and LinkedIn Learning offer Node.js courses that are structured to guide you through a specific learning path.

Ultimately, the most effective way to learn Node.js will depend on your individual learning style and preferences. It may be helpful to try a combination of tutorials, documentation, and courses to find the approach that works best for you.

**Comment by maahaMaya** [comment]
can u please have marked as answered.
